src/main/java/com/iyzipay/HashGenerator.java

```diff
@@ -22,4 +27,35 @@ public static String generateHash(String apiKey, String secretKey, String random
         }
         return DatatypeConverter.printBase64Binary(result);
     }
-}
+
+    public static String generateHashV2(String uri, String apiKey, String secretKey, String randomString, Object request) {
+        try {
+            String input = "apiKey:" + apiKey + "&randomKey:" + randomString + "&signature:" + getHmacSHA256Signature(uri, secretKey, randomString, request);
+            return DatatypeConverter.printBase64Binary(input.getBytes("UTF-8"));
+        } catch (UnsupportedEncodingException e) {
+            throw new HttpClientException("Hash cannot be generated", e);
```

Comment:
"Authentication Data cannot be generated" olsa daha güzel ismi. Burda yapılan sadece hash almak değil.

Reply:
katiliyorum; fakat mevcut entegrasyonlarda ```Hash cannot be generated``` uzerine kurulmus exception handling mekanizmalari olabilecegi icin bunu ayni biraktim.

Reply:
Bilemedim exception mesajına bağlı entegrasyon var mıdır. Var diyorsan kalsın da mesaj doğru değil.

Reply:
degistirdim buralari da ```IyziAuthenticationV2Generator``` class'ina tasininca. eski yaklasima uymamiza gerek kalmadi.